### ISSUE TITLE
Support apns-collapse-id Header

### ISSUE THREAD

**Issue by BlueHuang88** [alexalok/dotAPNS#57]
Hi,

I'd like to replace the content of notification ,then I need to add a "apns-collapse-id" field  to the header.
Are there any plans to implement it?

Thanks,

**Comment by alexalok** [comment]
Hi,

> I'd like to replace the content of notification

if you want to send a [mutable push](https://developer.apple.com/documentation/usernotifications/modifying_content_in_newly_delivered_notifications), you may use `ApplePush.AddMutableContent()`, i.e.

```c#
 var push = new ApplePush(ApplePushType.Alert)
    .AddMutableContent()
    .AddAlert("Title that can be replaced", "Body that can be replaced");
```

> I need to add a "apns-collapse-id" field to the header.

`apns-collapse-id`, on the other side, is used to coalesce multiple notifications into a single notification for the user, as mentioned in the [docs](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns/). This feature is indeed not supported at this moment.

If you're sure you definitely need an `apns-collapse-id` support, please comment on that and I'll add it to my backlog.

**Comment by BlueHuang88** [comment]
Thanks for you reply.

Sorry to make you misunderstand because of my unclear expression. I want to update content of notification that has sent to device. 

So I'm sure I need apns-collapse-id field, thanks for your help.

**Comment by alexalok** [comment]
Got it, I'll work on that in a free time. 

Still, I'd like to emphasize again that you probably need a [mutable push](https://developer.apple.com/documentation/usernotifications/modifying_content_in_newly_delivered_notifications) to update the content of notifications, which does not require `apns-collapse-id`. Instead, it requires `mutable-content: 1` key in the push payload itself which can be achieved by using the code snippet from my previous comment.

`apns-collapse-id`, on the other side, simply allows to combine multiple notifications. Think about messengers: multiple messages from one of the contacts can be grouped together so as to take less space on the notifications center.

**Comment by BlueHuang88** [comment]
Thanks for your help .

**Comment by BlueHuang88** [comment]
Hi 

I see the new version 3.3.0 , and download it from NuGet . But I don't see the change from this says https://github.com/alexalok/dotAPNS/commit/0742dc04257584640cd419a58d3a5244559c3409 .

I use WinMerge and it's similar to 3.2.0 . Could you please check on it ?

Thanks

**Comment by alexalok** [comment]
Oops, my bad! Fixed in [3.3.1](https://www.nuget.org/packages/dotAPNS/3.3.1).
